download/download.go

```diff
@@ -314,11 +316,15 @@ func (d *Downloader) download(ctx context.Context, m metrics.Metrics) (*download
 			baseURL := path.Join(d.client.Config().URL, d.path)
 
 			var loader bundle.DirectoryLoader
+
+			var dup bytes.Buffer
+			teeDup := io.TeeReader(resp.Body, &dup)
+
 			if d.persist {
-				tee := io.TeeReader(resp.Body, &buf)
+				tee := io.TeeReader(teeDup, &buf)
 				loader = bundle.NewTarballLoaderWithBaseURL(tee, baseURL)
 			} else {
-				loader = bundle.NewTarballLoaderWithBaseURL(resp.Body, baseURL)
+				loader = bundle.NewTarballLoaderWithBaseURL(teeDup, baseURL)
```

Comment:
I wonder if we can simplify this. Something like:

```go
tee := io.TeeReader(resp.Body, &buf)
loader = bundle.NewTarballLoaderWithBaseURL(tee, baseURL)
```

And then below we can return 

```go
return &downloaderResponse{
    b:        &b,
    raw:      &buf,
    etag:     etag,
    longPoll: isLongPollSupported(resp.Header),
    size:     buf.Len(), 
}, nil
```

Reply:
Good call. I think we can possibly even save another line:
`loader := bundle.NewTarballLoaderWithBaseURL(io.TeeReader(resp.Body, &buf), baseURL)`

Reply:
Done.